### PR TITLE
1761 objects settings add a cover image

### LINKED ISSUE #1761: Objects Settings - Add a cover image
# Behaviors

1. Add a LightIconButton "X" with position absolute over the cover image.
2. Allow users to discard the banner by clicking on the LightIconButton. The image should disappear with a 300ms fade-out animation. The content below (the SettingsTable block) should also scroll up with an animation.
3. Store the user's preference settings in either session storage or a cookie.

# Figma

https://github.com/twentyhq/twenty/assets/19412894/a62d145f-8343-4d4d-9825-b6fa2c844a1c

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11440-65418&mode=design&t=pzLEmoCvOdtrPWEL-4

Please note that it was not possible to insert the animations mentionned above into this Figma

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #1761 

- Added a cover image component
- Created the animation with framer motion in a reusable component called AnimatedFadeOut
- Created a cookie named `settings-object-cover-image` to store the user preference (cover image opened or closed)

Animation:

https://github.com/twentyhq/twenty/assets/71827178/a3eb3b02-9609-447a-9a17-6ce57cfc21e6

Cookie Behavior:

https://github.com/twentyhq/twenty/assets/71827178/77bef317-ffbc-4628-a412-925a1e3cde43

### CODE REVIEW

File: front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { LightIconButton } from '@/ui/input/button/components/LightIconButton';
+import { AnimatedFadeOut } from '@/ui/utilities/animation/components/AnimatedFadeOut';
+import { cookieStorage } from '~/utils/cookie-storage';
+
+import CoverImage from '../assets/build-your-business-logic.jpg';
+
+const StyledCoverImageContainer = styled.div`
+  border: 1px solid ${({ theme }) => theme.border.color.medium};
+  border-radius: ${({ theme }) => theme.border.radius.md};
+  height: 153px;
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx:15]: As we already have width: 100%, I think we can remove this height, the image should take the right height based on width x its ratio

File: front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  overflow: hidden;
+  position: relative;
+  width: 100%;
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx:18]: is this position necessary?

File: front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  return (
+    <AnimatedFadeOut
+      isOpen={cookieState !== 'closed'}
+      marginBottom={theme.spacing(8)}
+    >
+      <StyledCoverImageContainer>
+        <StyledCoverImage src={CoverImage} alt="Build your business logic" />
+        <StyledLighIconButton
+          Icon={IconX}
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx:48]: I'm a bit disappointed we go for IconX and not IconCross :D

File: front/src/modules/ui/utilities/animation/components/AnimatedFadeOut.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    <AnimatePresence>
+      {isOpen && (
+        <motion.div
+          initial={{
+            opacity: 1,
+            marginBottom: marginBottom ?? 0,
+            marginTop: marginTop ?? 0,
+          }}
+          exit={{ opacity: 0, height: 0, marginBottom: 0, marginTop: 0 }}
+          transition={{ duration: duration ?? 0.3, ease: 'easeOut' }}
```
Comment (charlesBochet) [inline front/src/modules/ui/utilities/animation/components/AnimatedFadeOut.tsx:28]: durations are part of the theme I think, they should be somewhere

File: front/src/modules/ui/utilities/animation/components/AnimatedFadeOut.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+type AnimatedFadeOutProps = {
+  isOpen: boolean;
+  children: React.ReactNode;
+  duration?: number;
+  marginBottom?: string;
+  marginTop?: string;
+};
+
+export const AnimatedFadeOut = ({
```
Comment (charlesBochet) [inline front/src/modules/ui/utilities/animation/components/AnimatedFadeOut.tsx:11]: this is great!

**Review by charlesBochet — APPROVED** [review]
Approved with minor comments

File: front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx (hunk 2 above)
Comment (bosiraphael) [reply front/src/modules/settings/data-model/objects/SettingsObjectCoverImage.tsx:18]: Yes, otherwise the absolute positioning of the child is incorrect

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**